compiler/rustc_codegen_llvm/src/llvm/ffi.rs

```diff
@@ -108,26 +133,36 @@ pub enum CallConv {
     AvrInterrupt = 85,
 }
 
-/// LLVMRustLinkage
-#[derive(Copy, Clone, PartialEq)]
+/// Must match the layout of `LLVMLinkage`.
+#[derive(Copy, Clone, PartialEq, TryFromU32)]
 #[repr(C)]
 pub enum Linkage {
```

Comment:
Remark: I double-checked this enum definition against https://github.com/llvm/llvm-project/blob/bb00f5b1edd0ed77b7e7b0113dad223505564b18/llvm/include/llvm-c/Core.h#L193, still seems to line up even on `main` (which hasn't changed like, since 11-12 years ago).

Reply:
Yeah, based on my archeology I think what happened is:
- Someone set up a Rust-side enum to mirror the C enum
- Later, the unstable C++ enum was changed (to remove deprecated values), but the C enum remained stable
- Someone updated the Rust enum to match the changes to the C++ enum, which was incorrect
- This broke things, and the C enum was incorrectly blamed for the breakage

Reply:
Yeah that seems most probable.